### ISSUE TITLE
Razzle and Lerna Monorepo

### ISSUE THREAD

**Issue by rayandrew** [jaredpalmer/razzle#664]
Hi, is there anyone here already implemented Razzle and Lerna?

I'm trying to splitting React UI-kit and containers into each self-package.

```bash

.
└── packages
    ├── client
    │   ├── build
    │   ├── public
    │   └── src
    ├── general-libs
    │   └── constants
    ├── server
    │   └── src
    └── ui-kit
        ├── config
        ├── public
        ├── scripts
        └── src

```

Client has dependencies to ui-kit and general-libs.
The problem is code not successfully compiled in server side razzle.

<img width="1310" alt="screen shot 2018-06-17 at 22 46 01" src="https://user-images.githubusercontent.com/4437323/41509703-1df8bcd0-7282-11e8-9691-e95bb822bf40.png">

I have already tried to add dependecy in client's package.json, include real-path of ui-kit in babel-loader rules, add resolve modules and also resolve loader.

I have already tried this [solution](https://github.com/lerna/lerna/issues/487) but it not works too.

Is there any workaround for this problems?

Thank you

**Comment by jaredpalmer** [comment]
Server and client entry need to share the same package.json. But you can alter the entry points however you want. Thus you’d have 3 packages instead of 4. Like app, ui, common. App would contain server and client

**Comment by rayandrew** [comment]
thanks for your answer!


but i'm sorry not explaining it nicely. server is different from 'server side' from razzle. its just another rest api server for data fetching. sorry for bad terminology.

thank you, i will try it

**Comment by mikestopcontinues** [comment]
@rayandrews did you get it working? I'm sniffing around after finding Lerna+NextJS impossible to work with. I can recommend using a babel.config.js file in the root of your repo and setting babel's rootMode = 'upward'. I was able to use this approach with other subprojects that import untranspiled sibling code. But I'd really like to be suer there's a way to set rootMode before I experiment porting everything from Next. I wasted a lot a time trying to get this working already.
